### PR TITLE
Time-over-time comparison aggregation helpers

### LINKED ISSUE #43079: Time-over-time comparison aggregation helpers

### PR DESCRIPTION
**Pull request by kamilmielnik**
Closes #43079

### Description

3 helper functions, as specified in [[Tech] Time-over-time comparison UI for aggregated columns](https://www.notion.so/metabase/Tech-Time-over-time-comparison-UI-for-aggregated-columns-d06c46884d924bb99b814ec9d8f9b114).
I decided to define them in a new file since these are not really low-level "core" functions and IMO should not clutter `expression.ts` (BTW these aren't really aggregation-specific, they will work with custom columns and filters in the future).
It's going straight to `master` since the functions aren't used yet.
Includes test coverage according to [these product specs](https://www.notion.so/metabase/Time-over-time-comparison-UI-for-aggregated-columns-032bb9b47d124731baa9c3c7f954727f).

### How to verify

New tests pass.
Manually check if assertions in new tests cover [product specs](https://www.notion.so/metabase/Time-over-time-comparison-UI-for-aggregated-columns-032bb9b47d124731baa9c3c7f954727f) - especially the aggregation names.

### CODE REVIEW

File: frontend/src/metabase-lib/expression.ts [hunk 1]
```diff
@@ -100,3 +104,66 @@ export function diagnoseExpression(
     expressionPosition,
   );
 }
+
+export function offsetClause(
+  query: Query,
+  stageIndex: number,
+  clause: AggregationClause | ExpressionClause,
```
Comment (kamilmielnik) [inline frontend/src/metabase-lib/expression.ts:111]: `offset` isn't supported in `FilterClause` yet

File: frontend/src/metabase-lib/offset.ts [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+  const bucket = temporalBucket(firstBreakout);
+
+  if (!bucket) {
+    return absoluteOffset === 1
+      ? t`${displayName} (${prefix}previous period)`
+      : t`${displayName} (${prefix}${absoluteOffset} periods ago)`;
+  }
+
+  const bucketInfo = displayInfo(query, stageIndex, bucket);
+  const period = inflect(bucketInfo.shortName, absoluteOffset);
```
Comment (kamilmielnik) [inline frontend/src/metabase-lib/offset.ts:106]: `inflect` will work only for the English language.

Awaiting product decision if we need to improve this or not: [Slack](https://metaboat.slack.com/archives/C0645JP1W81/p1716813789664009?thread_ts=1716475674.712849&cid=C0645JP1W81)

Comment (kamilmielnik) [reply frontend/src/metabase-lib/offset.ts:106]: > `inflect` will work only for the English language.
> 
> Awaiting product decision if we need to improve this or not: [Slack](https://metaboat.slack.com/archives/C0645JP1W81/p1716813789664009?thread_ts=1716475674.712849&cid=C0645JP1W81)

[It'll do](https://metaboat.slack.com/archives/C0645JP1W81/p1716824103084089?thread_ts=1716475674.712849&cid=C0645JP1W81) for now.

**Review by romeovs — APPROVED** [review]
Looks good!

**Pull request merged**